stdlib/src/builtin/string.mojo

```diff
@@ -51,6 +51,25 @@ fn _ctlz(val: SIMD) -> __type_of(val):
     return llvm_intrinsic["llvm.ctlz", __type_of(val)](val, False)
 
 
+@always_inline
+fn unicode_compare[T: Stringable](str1: T, str2: T) -> Int:
```

Comment:
This is an internal function, maybe prefix the name with `_`. Also, I think we should restrict the input type to `String`.